app/helpers/pdf_helper.rb

```diff
@@ -3,15 +3,28 @@ def page_break
     content_tag(:div, "", class: "page-break")
   end
 
-  def asset_data_uri(path)
-    asset = Rails.application.assets.find_asset(path)
-    data = Base64.encode64(asset.source)
-    "data:#{asset.content_type};base64,#{Rack::Utils.escape(data)}"
+  def asset_data_uri(asset)
+    asset_data = get_asset(asset)
+    data = Base64.encode64(asset_data.source)
+    "data:#{asset_data.content_type};base64,#{Rack::Utils.escape(data)}"
   end
 
   def pdf_style_sheet
+    asset_data = get_asset("pdf.css")
     content_tag(:style, type: "text/css") do
-      Rails.application.assets.find_asset("pdf.scss").to_s.html_safe # rubocop:disable Rails/OutputSafety
+      asset_data.source.html_safe # rubocop:disable Rails/OutputSafety
+    end
+  end
+
+  def get_asset(asset)
+    if Rails.application.config.assets.compile
```

Comment:
Does this compile all the assets manually each time this method is called?  That seems very time-consuming.

Reply:
No it's just a boolean variable (indicating whether or not Sprockets is around to compile assets for you).